code/messaging/src/phone/java/com/adobe/marketing/mobile/Messaging.java

```diff
@@ -218,12 +218,14 @@ public static void setPropositionsHandler(@NonNull final AdobeCallback<Map<Surfa
                 final Map<String, Object> eventData = event.getEventData();
                 if (!MapUtils.isNullOrEmpty(eventData)) {
                     final List<Map<String, Object>> retrievedPropositionData = DataReader.optTypedListOfMap(Object.class, eventData, PROPOSITIONS, Collections.emptyList());
-                    for (final Map<String, Object> propositionData : retrievedPropositionData) {
-                        final Surface surface = Surface.fromUriString(DataReader.optString(propositionData, SCOPE, null));
-                        final List<Proposition> propositions = new ArrayList<>();
-                        final Proposition proposition = Proposition.fromEventData(propositionData);
-                        propositions.add(proposition);
-                        convertedPropositions.put(surface, propositions);
+                    if (retrievedPropositionData != null && !retrievedPropositionData.isEmpty()) {
+                        for (final Map<String, Object> propositionData : retrievedPropositionData) {
+                            final Surface surface = Surface.fromUriString(DataReader.optString(propositionData, SCOPE, null));
+                            final List<Proposition> propositions = new ArrayList<>();
+                            final Proposition proposition = Proposition.fromEventData(propositionData);
+                            propositions.add(proposition);
+                            convertedPropositions.put(surface, propositions);
```

Comment:
All propositions for a given scope need to be accumulated in the list before being added in the map (to be returned) for the surface.